### PR TITLE
Incorporate data_holder into the reading of files.

### PR DESCRIPTION
**Pull request by rphilip457**
This pull request builds the ability to combine the data_holder and validator functionality in the reading of files.

 - Moved Data class to using kwargs with the keywords being the column names. This means that the columns can be out of order and the reading will still work. Suggest that initialization of data class is done with a dictionary using the ** prefix to turn it into kwargs.
 - Added a function called reading_data that handles a Data input rather than columns
 - Added a function called construct_datacontainer that holds the read in data.

### CODE REVIEW

File: data_holder.py [hunk 1]
```diff
@@ -48,4 +48,4 @@ class Name(Data):
             "Column 2": Number(minvalue=1, maxvalue=100),
             "Column 3": Number(minvalue=1, maxvalue=100),
         }
-    name_row = Name(1, 1, 1)
+    name_row = Name(**{"Column 1" : 1, "Column 2" : 1, "Column 3" : 1})
```
Comment (jTaylo99) [inline data_holder.py]: Very minor thing, for Python dicts generally you don't have a space between the key and the colon, eg. {"Key": Value}

File: kolumnextractor.py [hunk 2]
```diff
@@ -13,14 +13,26 @@ def filter_columns(raw_data: list, columns_to_filter):
     return [[row[index] for index in index_to_keep] for row in raw_data]
 
 
+def construct_datacontainer(raw_data: list, columns: Data):
+    data = []
+    for r in raw_data:
+        row = {col: r[col] for col in columns._columns.keys()}
```
Comment (jTaylo99) [inline kolumnextractor.py:19]: Getting an error here. The issue is regarding how our column names have no space in our Data object, but the raw data has spaces causing a key error.

File: kolumnextractor.py [hunk 3]
```diff
@@ -13,14 +13,26 @@ def filter_columns(raw_data: list, columns_to_filter):
...
+    return DataContainer(data)
+
+
 def read_csv(info_dict):
     with open(info_dict["path"], newline='') as csvfile:
         reader = csv.reader(csvfile, delimiter=',', quotechar='|')
         all_items = list(reader)
+        headers = all_items.pop(0)
+        raw_data = [dict(zip(headers, r)) for r in all_items]
+        print(raw_data)
```
Comment (jTaylo99) [inline kolumnextractor.py]: Leftover print statement

**Review by jTaylo99 — CHANGES_REQUESTED** [review]

**Comment by rphilip457** [comment]
Have fixed the problems you have requested changed. But have uncovered another bug in doing so. Currently we read in the rows from the csv as strings. We are then using the validation to check that the numbers are `int` or `float` but they are never set to int or float so this will always fail. How would you like to handle this? 

We could check what characters the string contains and if it meets requirements we can parse it into the correct type and if not then throw the type error?

**Comment by jTaylo99** [comment]
Good catch! I'd say your idea is a good solution. This could be another PR as this one is already doing quite a bit

**Review by jTaylo99 — APPROVED** [review]

**Comment by jTaylo99** [comment]
I'm happy with the changes you've made. Right now the package is in a state where it's not fully ready for use so I'm happy to push it through with the bug you've found - but I will add an issue for said bug as it's not one that was caused by you.

**Pull request merged**